pkg/i2gw/intermediate/provider_ingressnginx.go

```diff
@@ -16,6 +16,24 @@ limitations under the License.
 
 package intermediate
 
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
 type IngressNginxGatewayIR struct{}
-type IngressNginxHTTPRouteIR struct{}
+
+type IngressNginxHTTPRouteIR struct {
```

Comment:
I think we should refactor the way we are doing provider-specific extensions. That is, we should get rid of as many of them as possible. Right now, only GCE uses them and we could keep them around for backcompatibility if we wanted. Ideally, any IR passed to the emitter would have no reference to the provider (or any ingress-related resource).